libs/server-sent-events/src/client.cpp

```diff
@@ -141,11 +149,17 @@ class FoxyClient : public Client,
         if (ec == boost::asio::error::operation_aborted) {
             return;
         }
-        run();
+        do_run();
     }
 
-    void run() override {
-        session_.async_connect(
```

Comment:
After renaming `run`, I wrapped it in a `post` so we can ensure there is only one thread accessing `session`. This is the same as `async_shutdown`.